app/filtering.js

```diff
@@ -530,7 +530,7 @@ function registerPermissionHandler (session, partition) {
           response[index] = result
           if (persist) {
             // remember site setting for this host
-            appActions.changeSiteSetting(origin, permission + 'Permission', result, isPrivate)
+            appActions.changeSiteSetting(mainFrameUrl, permission + 'Permission', result, isPrivate)
```

Comment:
I don't think we want to change this behavior for permissions in general - ex: if you go to https://developer.mozilla.org/en-US/docs/Web/API/notification and click 'notify me', the message is `Allow https://mdn.mozillademos.org...` instead of `Allow https://developer.mozilla.org...`. we would also have to clear all saved permissions for this to work correctly. 

for https://github.com/brave/browser-laptop/issues/13471, i think the most minimal fix is to change the message to "Allow X to be opened in an external app?" for the `openExternal` permission only and set that permission by URL about to be opened instead of the current page URL